src/transformations/execute.ts

```diff
@@ -232,6 +236,47 @@ export async function executeTransformations(
         vimState.editor.selection = new vscode.Selection(newPos, newPos);
         break;
 
+      case 'executeNormal':
+        const keystroke = keystrokesExpressionParser.parse(transformation.keystroke);
+        const startLineNumber = transformation.startLineNumber;
+        const endLineNumber = transformation.endLineNumber;
+        const withRange = transformation.withRange;
+        if (!keystroke.status) {
+          throw new Error(`Failed to execute normal command: ${transformation.keystroke}`);
+        }
+
+        const resultLines: TextLine[] = [];
```

Comment:
We collect these `TextLine`s only to use them for their line numbers? If that's the case, just store the line numbers directly, as `TextDocument.lineAt()` is fairly cheap until someone runs `:%normal` in a big file.

Reply:
Fix it.
[283e7c6](https://github.com/VSCodeVim/Vim/pull/8896/commits/283e7c66467b63c191de99d4e4c3d0e560368fc2)